src/box/memtx_tree.cc

```diff
@@ -365,27 +388,20 @@ tree_iterator_next_equal_base(struct iterator *iterator, struct tuple **ret)
 	} else {
 		memtx_tree_iterator_next(&index->tree, &it->tree_iterator);
 	}
-	tuple_unref(it->current.tuple);
 	struct memtx_tree_data<USE_HINT> *res =
 		memtx_tree_iterator_get_elem(&index->tree, &it->tree_iterator);
+	struct index *idx = iterator->index;
+	struct space *space = space_by_id(iterator->space_id);
 	/* Use user key def to save a few loops. */
 	if (res == NULL ||
 	    tuple_compare_with_key(res->tuple, res->hint,
 				   it->key_data.key,
 				   it->key_data.part_count,
 				   it->key_data.hint,
 				   index->base.def->key_def) != 0) {
+		tree_iterator_set_current<USE_HINT>(it, NULL);
```

Comment:
May be use `tree_iterator_set_current_tuple` here to prevent changing hint as it was previously

Reply:
After the next commit, `tree_iterator_set_current`, apart from dereferencing the tuple, will also free the functional index "chunk" so using `tree_iterator_set_current_tuple` wouldn't be enough.